pkg/virt-controller/watch/vmi.go

```diff
@@ -1115,6 +1129,12 @@ func (c *VMIController) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod,
 			log.Log.V(3).Object(vmi).Infof("Delaying pod creation while DataVolume populates")
 			return nil
 		}
+
+		// ensure that the backend storage associated with the VMI is ready before creating the pod
+		if !backendStorageReady {
```

Comment:
Currently, `backendstorage.CreateIfNeeded` always returns either `true, nil` or `false, err` (if `err != nil`). Is there really a case when `backendStorageReady` will result in `false` and we get with delaying pod creation? Would it make sense to simplify `CreateIfNeeded` and make it simply return an error?